SampleApplications/SDK/Opc.Ua.Client/CoreClientUtils.cs

```diff
@@ -35,24 +35,43 @@ namespace Opc.Ua.Client
     /// <summary>
     /// Defines numerous re-useable utility functions for clients.
     /// </summary>
-    public class CoreClientUtils
+    public static class CoreClientUtils
     {
+        /// <summary>
+        /// The default discover operation timeout.
+        /// </summary>
+        public static readonly int DefaultDiscoverTimeout = 15000;
+
         #region Discovery
         /// <summary>
         /// Discovers the servers on the local machine.
         /// </summary>
         /// <param name="configuration">The configuration.</param>
         /// <returns>A list of server urls.</returns>
         public static IList<string> DiscoverServers(ApplicationConfiguration configuration)
+        {
+            return DiscoverServers(configuration, DefaultDiscoverTimeout);
+        }
+
+        /// <summary>
+        /// Discovers the servers on the local machine.
+        /// </summary>
+        /// <param name="configuration">The configuration.</param>
+        /// <param name="discoverTimeout">Operation timeout in milliseconds.</param>
+        /// <returns>A list of server urls.</returns>
+        public static IList<string> DiscoverServers(
+            ApplicationConfiguration configuration,
+            int discoverTimeout
+            )
         {
             List<string> serverUrls = new List<string>();
 
             // set a short timeout because this is happening in the drop down event.
-            EndpointConfiguration endpointConfiguration = EndpointConfiguration.Create(configuration);
-            endpointConfiguration.OperationTimeout = 5000;
+            var endpointConfiguration = EndpointConfiguration.Create(configuration);
+            endpointConfiguration.OperationTimeout = discoverTimeout;
 
             // Connect to the local discovery server and find the available servers.
-            using (DiscoveryClient client = DiscoveryClient.Create(new Uri("opc.tcp://localhost:4840"), endpointConfiguration))
+            using (DiscoveryClient client = DiscoveryClient.Create(new Uri(String.Format(Utils.DiscoveryUrls[0], "localhost")), endpointConfiguration))
```

Comment:
>String.Format(Utils.DiscoveryUrls[0], "localhost") [](start = 75, length = 50)

[Quality] Format string is not necessary

Reply:
yet unclear to me why not, but can keep it for later

---
In reply to: [432603758](https://github.com/OPCFoundation/UA-.NETStandard/pull/998#discussion_r432603758) [](ancestors = 432603758)